CppCoreGuidelines.md

```diff
@@ -3922,21 +3922,27 @@ value) of any assignment operator.
 
 ##### Reason
 
-With guaranteed copy elision, it is now almost always a pessimization to expressly use `std::move` in a return statement.
+Returning a local variable implicitly moves it anyway.
+An explicit `std::move` is always a pessimization, because it prevents Return Value Optimization (RVO),
+which can eliminate the move completely. 
 
 ##### Example, bad
 
-    S f()
+    S bad()
     {
       S result;
       return std::move(result);
     }
 
 ##### Example, good
 
-    S f()
+    // RVO: guaranteed move elision when a temporary is returned
```

Comment:
language lawyers will argue that C++17 prvalue return is not an optimization and so not "RVO"
Perhaps omit it and just focus on returning local variables?

Reply:
You're right, the standard doesn't call it RVO or NRVO, it only says that returning is *copy-initialization* of the returned object. Nested prvalue initializations like `T(T(T(T(...)))))` are just elided to one `T()`.

I think it would be best to remove the explicit mention of RVO and NRVO from the diff, but keep the rest of the comment.